android_emulator_cli.sh

```diff
@@ -93,46 +128,121 @@ fi
 export ANDROID_SDK_ROOT=~/Library/Android/sdk
 export ANDROID_HOME=~/Library/Android/sdk
 export PATH=$PATH:$ANDROID_HOME/cmdline-tools/latest/bin/
+
+# Android CLI Tools END
+########################################################
+# SDKManager Platform-Tools & Emulator START
+
 echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
 echo "Ensuring that you are updated to the latest version."
 echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
 echo "Downloading Android Platform Tools and Emulator files."
+
 yes | ~/Library/Android/sdk/cmdline-tools/latest/bin/sdkmanager platform-tools emulator
+
 echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
 echo ""
+echo ""
 echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
-echo "Downloading Android API 29 files."
-yes | ~/Library/Android/sdk/cmdline-tools/latest/bin/sdkmanager "platforms;android-29"
+
+sleep .5
+echo "Getting Recommended Android API"
+
+ANDROID_API=$(~/Library/Android/sdk/cmdline-tools/latest/bin/sdkmanager --list --channel=0 | grep -o 'platforms;android-[0-9][0-9]' | awk -F"android-" '/android-/{ print $2}' | tail -2 | head -1)
+
```

Comment:
From what I gathered there is no "latest" version other than for the `command line tools`. And because we don't immediately target the latest API version, I figured getting the second latest API version is the safest path forward.

What this script does is:
* Lists out the production versions (`channel=0` is `production`) of `platforms;android-<api version>`
* Gets the version number by removes the extra text in the returned list with `awk -F"android-" '/android-/{ print $2}'
* Displays only the last 2 entries `tail -2` (ie. API's `32` & `33`)
* Then displays only then first entry from this list (ie. API `32`)

This value is stored locally to use later.